src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/IImportCompletionCacheService.cs

```diff
@@ -3,15 +3,18 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Generic;
+using System.Threading;
 using Microsoft.CodeAnalysis.Host;
 
-namespace Microsoft.CodeAnalysis.Completion.Providers.ImportCompletion
+namespace Microsoft.CodeAnalysis.Completion.Providers
 {
     internal interface IImportCompletionCacheService<TProject, TPortableExecutable> : IWorkspaceService
     {
         // PE references are keyed on assembly path.
         IDictionary<string, TPortableExecutable> PEItemsCache { get; }
 
         IDictionary<ProjectId, TProject> ProjectItemsCache { get; }
+
+        CancellationToken DisposalToken { get; }
```

Comment:
All the change in the last commits are refactoring except this and its implementation.